src/Networking/NexusMods.Networking.Downloaders/Tasks/State/DownloaderState.cs

```diff
@@ -0,0 +1,117 @@
+
+namespace NexusMods.Networking.Downloaders.Tasks.State;
+
+/// <summary>
+/// Stores the state of a suspended download.
+/// </summary>
+/// <remarks>
+///     <see cref="ITypeSpecificState"/> contains injected data.
+///     That is, state specific to the owner.
+///
+///     To give an example, <see cref="HttpDownloadTask"/> will have a <see cref="HttpDownloadState"/> injected; which
+///     stores an URL. That is specific to the <see cref="HttpDownloadTask"/>, and a task such as <see cref="NxmDownloadTask"/>
+///     might not need it since it uses another strategy to start a mod download.
+/// </remarks>
+// ReSharper disable once PartialTypeWithSinglePart
+[JsonName("NexusMods.Networking.Downloaders.Tasks.State.DownloaderState")]
+public record DownloaderState : Entity
+{
+    /// <summary>
+    /// Data tied to the type that created this state snapshot.
+    /// </summary>
+    public ITypeSpecificState? TypeSpecificData { get; private set; }
+
+    /// <summary>
+    /// Status of the task associated with this state.
+    /// </summary>
+    public DownloadTaskStatus Status { get; private set; }
+
+    /// <summary>
+    /// Path to the temporary file being downloaded.
+    /// </summary>
+    /// <remarks>
+    ///     In our code this is <see cref="AbsolutePath"/> from <see cref="TemporaryPath"/>; but we can't use that here
+    ///     for serialization reasons. Because <see cref="IFileSystem"/> always is <see cref="FileSystem"/> however,
+    ///     (more specifically, saved to `/tmp` i.e. `%temp%` on Windows) this is okay to store.
+    /// </remarks>
+    public string DownloadPath { get; private set; } = "";
+
+    /// <summary>
+    /// Friendly name for the suspended download task.
+    /// </summary>
+    public string FriendlyName { get; private set; } = "";
+
+    /// <summary>
+    /// Amount of already downloaded bytes.
+    /// </summary>
+    public long DownloadedBytes { get; private set; } = 0L;
+
+    /// <summary>
+    /// Name of the game the mod will be installed to.
+    /// </summary>
+    /// <remarks>Provided by <see cref="IHaveGameName"/> trait.</remarks>
+    public string? GameName { get; private set; }
+
+    /// <summary>
+    /// Unique identifier for the game whose loadouts should be suggested for installation.
+    /// </summary>
+    /// <remarks>Provided by <see cref="IHaveGameDomain"/> trait.</remarks>
+    public string? GameDomain { get; private set; }
+
+    /// <summary>
+    /// Size of the file being downloaded in bytes. A value of less than 0 means size is unknown.
+    /// </summary>
+    /// <remarks>Provided by <see cref="IHaveFileSize"/> trait.</remarks>
+    public long? SizeBytes { get; private set; }
+
+    /// <summary>
+    /// Version of the mod; can sometimes be arbitrary and not follow SemVer or any standard.
+    /// </summary>
+    /// <remarks>Provided by <see cref="IHaveDownloadVersion"/> trait.</remarks>
+    public string? Version { get; private set; }
```

Comment:
if we don't set these members outside of the `Create` we should be using `required` and `init` vs `private set`. This helps enforce immutability.

Reply:
Resolved in 8180e754007c1696c438a84853f162f171bad718